.github/workflows/extract-api.yaml

```diff
@@ -17,30 +17,22 @@ jobs:
   extract-api:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout branch
-        uses: actions/checkout@v2
+      - uses: actions/checkout@v4
 
-      - name: Configure git
-        run: |
-          git config --local user.email imodeljs-admin@users.noreply.github.com
-          git config --local user.name imodeljs-admin
-
-      - name: Setup node
-        uses: actions/setup-node@v4
+      - uses: pnpm/action-setup@v3
         with:
-          node-version: 20.x
-
-      - name: Rush install
-        run: node common/scripts/install-run-rush.js install
+          version: 10
 
-      - name: Rush build
-        run: node common/scripts/install-run-rush.js build -v -p max --to-version-policy prerelease-monorepo-lockStep
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: "pnpm"
 
-      - name: Rush extract-api
-        run: node common/scripts/install-run-rush.js extract-api
+      - run: pnpm install
+      - run: pnpm build
+      - run: pnpm extract-api
 
-      - name: Publish extracted api
-        uses: actions/upload-artifact@v4
+      - uses: actions/upload-artifact@v4
```

Comment:
What is happening with uploaded artifact? Maybe this job could be moved to `CI.yaml` so it could reuse cached `node_modules`?